### ISSUE TITLE
toPythonModule(toPythonApplication) could add(remove) python version prefixes in the store path

### ISSUE THREAD

**Issue by veprbl** [NixOS/nixpkgs#89593]
`toPythonApplication` seems to imply this:
https://github.com/NixOS/nixpkgs/blob/5f0bd4d70f18b3499005eb61a34acb98c9b05756/pkgs/top-level/python-packages.nix#L90-L93
But it doesn't work as it sets `passthru.name`:
```
nix-repl> (import <nixpkgs> {}).sqlmap  
«derivation /nix/store/6h6gd086bz4z85knyamnlkh3p1qvrvxy-python3.7-sqlmap-1.4.5.drv»

nix-repl> (import <nixpkgs> {}).sqlmap.name
"sqlmap-1.4.5"
```

The benefit of the current approach is that it may save some rebuilds in some cases. In case of `toPythonModule` many packages may need to be rebuilt because of `enablePython = true` settings. And, presumably, builds that go into `toPythonApplication` are not very heavy ones.

**Comment by FRidh** [comment]
Yes, adding it to `passthru.name` was on purpose to avoid the rebuild. 

Was this a question, or do you think something should be changed?

**Comment by veprbl** [comment]
> Yes, adding it to `passthru.name` was on purpose to avoid the rebuild.

Then we at very least need to correct the comment: `passthru.name` does not change the store path.

> Was this a question, or do you think something should be changed?

I'm not entirely sure if this needs to be changed, but it seems like it only avoids rebuilds in a minority of cases. Here is a crude way to check for this:
```
cat pkgs/top-level/python-packages.nix | awk -v ORS= '/=.*toPythonModule/ { print "(with import ./. {}; " $1 " == python3Packages." $1 ")\0"; }' | xargs -0 -n1 nix eval
```
The way I see it, we could commit either to having python version prefixes clearly visible in the store paths, or we could commit to reducing rebuilds. Right now we only do a little bit of the latter.
